readthedocs/proxito/middleware.py

```diff
@@ -22,6 +22,59 @@
 log = structlog.get_logger(__name__)  # noqa
 
 
+def _get_domain_from_host(host):
+    """
+    Get the normalized domain from a hostname.
+
+    A hostname can include the port.
+    """
+    return host.lower().split(":")[0]
+
+
+def _unresolve_domain(domain):
```

Comment:
It would be good if this function logs "where did we get the project slug from?". Was it from a subdomain of public domain? a custom domain? a subdomain of the external domain?

It could be in info mode for now while we are working on this refactor, so we can check that things are working correctly in production. Then, we can lower it down to debug.

Reply:
These are also logged outside the function https://github.com/readthedocs/readthedocs.org/blob/185771d2fabcdefa6519698d1e43e82e653bcad6/readthedocs/proxito/middleware.py#L125-L125, similar to https://github.com/readthedocs/readthedocs.org/pull/9462#discussion_r941537683 I wasn't sure if we should logs this outside or inside the function, also, they are set as debug as the old logs.

If you think this makes more sense inside the function, let me know.

Reply:
I think we should move the logic into this function, since this function will also be called in other places other than proxito right?